gateway/server.go

```diff
@@ -141,11 +141,12 @@ func main() {
 
 	r.HandleFunc("/system/alert", faasHandlers.Alert)
 
-	r.HandleFunc("/system/function/{name:[-a-zA-Z_0-9]+}", queryFunction).Methods(http.MethodGet)
+	r.HandleFunc("/system/function/{name:[-a-zA-Z_0-9]+}", queryFunction).
+		Methods(http.MethodGet, http.MethodHead, http.MethodOptions)
 	r.HandleFunc("/system/functions", faasHandlers.ListFunctions).Methods(http.MethodGet)
 	r.HandleFunc("/system/functions", faasHandlers.DeployFunction).Methods(http.MethodPost)
 	r.HandleFunc("/system/functions", faasHandlers.DeleteFunction).Methods(http.MethodDelete)
-	r.HandleFunc("/system/functions", faasHandlers.UpdateFunction).Methods(http.MethodPut)
+	r.HandleFunc("/system/functions", faasHandlers.UpdateFunction).Methods(http.MethodPut, http.MethodPatch)
```

Comment:
The back-ends don't understand what MethodPatch should do, so I think we should omit this from the scope of the PR.